contracts/exchangeIssuance/ExchangeIssuance.sol

```diff
@@ -0,0 +1,727 @@
+     * @param _minSetReceive    Minimum amount of index to receive
+     * @return setTokenAmount   Amount of set tokens issued
+     */
+    function _issueSetForExactWETH(ISetToken _setToken, uint256 _minSetReceive) internal returns (uint256) {
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        
+        ISetToken.Position[] memory positions = _setToken.getPositions();
+        
+        (uint256[] memory amountEthIn, Exchange[] memory exchanges, uint256 sumEth) = _getAmountETHForIssuance(_setToken);
+
+        uint256 setTokenAmount = _acquireComponents(positions, amountEthIn, exchanges, wethBalance, sumEth);    // acquire set components
+        
+        require(setTokenAmount > _minSetReceive, "ExchangeIssuance: INSUFFICIENT_OUTPUT_AMOUNT");
+        
+        basicIssuanceModule.issue(_setToken, setTokenAmount, msg.sender);       // issue token
+        
+        return setTokenAmount;
+    }
+    
+    /**
+     * Issues an exact amount of set tokens using WETH. 
+     * Acquires set token components at the best price accross uniswap and sushiswap.
+     * Uses the acquired components to issue the set tokens.
+     * 
+     * @param _setToken          Address of the set token being issued
+     * @param _amountSetToken    Amount of set tokens to issue
+     * @return sumEth            Total amount of ether used to acquire the set token components    
+     */
+    function _issueExactSetFromWETH(ISetToken _setToken, uint256 _amountSetToken) internal returns(uint256) {
+        
+        uint256 sumEth = 0;
+        
+        ISetToken.Position[] memory positions = _setToken.getPositions();
+        for(uint256 i = 0; i < positions.length; i++) {     // acquire set components
+            
+            uint256 amountToken = uint256(positions[i].unit).mul(_amountSetToken).div(1 ether);
+            
+            (, Exchange exchange) = _getMinTokenForExactToken(amountToken, WETH, positions[i].component);
+            uint256 amountEth = _swapTokensForExactTokens(exchange, WETH, positions[i].component, amountToken);
+            sumEth = sumEth.add(amountEth);
+        }
+        
+        basicIssuanceModule.issue(_setToken, _amountSetToken, msg.sender);      // issue token
+        
+        return sumEth;
+     }
+    
+    /**
+     * Handles converting the contract's full WETH balance to the output
+     * token or ether and transfers it to the msg sender.
+     *
+     * @param _isOutputETH      Converts the contract's WETH balance to ETH if set to true
+     * @param _outputToken      The token to swap the contract's WETH balance to. 
+     *                          Ignored if _isOutputETH is set to true.
+     * @param _minOutputReceive Minimum amount of output token or ether to receive. This 
+     *                          function reverts if the output is less than this. 
+     * @return                  Amount of output ether or tokens sent to msg.sender
+     */
+    function _handleRedeemOutput(
```

Comment:
Please remove this function, and handle the redeem output in the respective functions.